tests/api2/test_core_download.py

```diff
@@ -0,0 +1,29 @@
+from os import stat
+
+from urllib.request import urlretrieve
+from middlewared.test.integration.utils.client import truenas_server
+from middlewared.test.integration.utils import call
+
+
+def test_get_download_for_config_dot_save():
+    # ping core for sanity
+    assert call('core.ping') == 'pong'
+
+
+    # set up core download
+    job_id, url = call('core.download', 'config.save', [], 'freenas.db')
+
+
+    # is the job running?
+    results = call('core.get_jobs', [['id', '=', job_id]], {"get": True})
```

Comment:
This is a race. Starting the job and then asserting that's running is prone to failures.